### ISSUE TITLE
Map rating system

### ISSUE THREAD

**Issue by raptor** [bitfighter/bitfighter#143]
_From [watusim...@bitfighter.org](https://code.google.com/u/105427273526970468779/) on November 23, 2011 09:48:16_

Implement a map rating system.  Maps can be rated from 1 to 5 stars, and users can provide brief comments.  Comments and ratings will be stored on the master server, and will be available via web interface. 

Only registered players will be able to vote/comment, and they will be able to make and revise their ratings/comments in game via menu option.  

Maps will be indexed via md5 hash of the level file, and we should save details such as map name and creator on the master with the ratings for display on the web page.  Perhaps we should also include some sort of graphical rendering of the map, either a server-generated png, or some sort of conversion to svg that can be done without a screen context.

Also, map ratings will be displayed to all players when the level is loaded, if the master is available.

Finally, we can have a special section of the website for highest rated maps that have been rated by a minimum of, say, 5 registered players, and could award achievements based on having highly rated maps if we ever implement an achievement system.

This project will require changes to the protocol, so is not suitable for a 16a release, and is too large to accomplish in time for the scheduled release of 016.

_Original issue: http://code.google.com/p/bitfighter/issues/detail?id=143_

**Comment by raptor** [comment]
_From [watusim...@bitfighter.org](https://code.google.com/u/105427273526970468779/) on October 07, 2012 14:48:03_

I'm currently thinking just a +1, 0, -1 rating would suffice.

**Labels:** -017 019

**Comment by raptor** [comment]
_From [watusim...@bitfighter.org](https://code.google.com/u/105427273526970468779/) on November 26, 2012 03:38:02_

Notes from case 183, which will be closed as a duplicate

Let players rate levels with -1, 0, +1.  Ratings are to be retained and could be displayed in the corner of the game window when an already-rated level is active.  Players can change their rating of the current level if they have rated it in the past.

Cumulative/average ratings will be available on the server, and perhaps in the UI.  Might be better to display the rating as a number from 0-100, where 50 == 0 , 0 == -1, and 100 = 1.

Probably need a new key to be bound to toggle rating from -1 to 0 to +1

**Comment by raptor** [comment]
_From [watusim...@bitfighter.org](https://code.google.com/u/105427273526970468779/) on November 26, 2012 03:38:24_

Issue 183 has been merged into this issue.

**Comment by raptor** [comment]
_From [bkcon...@gmail.com](https://code.google.com/u/112584966209893525339/) on July 27, 2013 20:28:48_

**Labels:** -Type-Defect Type-Enhancement

**Comment by raptor** [comment]
_From [bkcon...@gmail.com](https://code.google.com/u/112584966209893525339/) on August 01, 2013 11:02:12_

**Labels:** 020

**Comment by raptor** [comment]
_From [bkcon...@gmail.com](https://code.google.com/u/112584966209893525339/) on August 01, 2013 11:05:14_

**Labels:** -019

**Comment by raptor** [comment]
_From [bkcon...@gmail.com](https://code.google.com/u/112584966209893525339/) on August 01, 2013 11:14:24_

\> be able to make and revise their ratings/comments in game via menu option

Currently, comments have no interface in the game. Rating requires the use of `/rate up` etc

\> Also, map ratings will be displayed to all players when the level is loaded, if the master is available.

The rating is printed only after `/rate` succeeds.

\> Probably need a new key to be bound to toggle rating from -1 to 0 to +1

I agree, perhaps Ctrl + F4/F5

**Labels:** -020 019

**Comment by raptor** [comment]
_From [watusim...@bitfighter.org](https://code.google.com/u/105427273526970468779/) on October 31, 2013 06:57:14_

I think this is done enough for 019; I'll open separate issues for what I feel are the outstanding points left to be done.

**Status:** Fixed
